api/v1alpha1/linodecluster_types.go

```diff
@@ -40,6 +40,11 @@ type LinodeClusterSpec struct {
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
 	// +optional
 	VPCRef *corev1.ObjectReference `json:"vpcRef,omitempty"`
+
+	// CredentialsRef is a reference to a Secret that contains the credentials to use for provisioning this cluster. If not
+	// supplied then the credentials of the controller will be used.
+	// +optional
+	CredentialsRef *corev1.SecretReference `json:"credentialsRef,omitempty"`
```

Comment:
Since `LinodeCluster` and others are namespace-scoped, should the type here just be a string and expect the secret exist in the same namespace?

Reply:
I think it could, but I wanted to leave the option open to use Secrets from a different namespace (like the GCP version). I'll defer to the team if we want to strictly enforce same-namespace Secrets.

I was planning to have the manifest templating default the `SecretReference.Namespace` value to where the cluster is installed though.

Reply:
Now that you mention it, I could see the use-case of referencing the same secret in some other namespace and not needing to propagate it everywhere. We'll adopt this convention as well in our ticket.